### PR TITLE
[ADD] [15.0]account_invoice_date

### PR DESCRIPTION
**Pull request by ikapasi-initos**

### CODE REVIEW

**Review by dsolanki-initos — APPROVED** [review]
Code review and functional test.
LGTM!

**Comment by AaronHForgeFlow** [comment]
Commit history is missing. Can you follow the steps described here https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-15.0 ?

**Comment by antoniocanovas** [comment]
Is it a migration or a new one?
I can't find it in previous versions...

**Comment by AaronHForgeFlow** [comment]
If this is a new module, can you please remove the word "migration" from the title here as well? thanks :)

**Comment by rafaelbn** [comment]
Hello @ikapasi-initos , I'm curious about why you this this module. Is this a UX button?

> #. Go to Invoice > Click Create > Create new invoice.
> #. Click on "Confirm" button.
> #. Select the date and Click on "Open invoice" button.

**Comment by jans23** [comment]
Yes, it's a button which allows to choose the invoice date when confirming the invoice. This helps when the company uses "actual taxation" (instead of "imputed taxation") which is not supported by Odoo directly.

**Comment by ikapasi-initos** [comment]
Hello @dalonsod it would be great if you add reviews here.

File: account_invoice_date/README.rst
Comment (dalonsod) [inline account_invoice_date/README.rst:1]: This file should not be included, it'll be autogenerated later

File: account_invoice_date/i18n/de.po
Comment (dalonsod) [inline account_invoice_date/i18n/de.po:1]: Please remove this, translations should be added using Weblate

File: account_invoice_date/models/__init__.py
Comment (dalonsod) [inline account_invoice_date/models/__init__.py]: These headers are actually unncessary in python files

File: account_invoice_date/security/ir.model.access.csv
Comment (dalonsod) [inline account_invoice_date/security/ir.model.access.csv]: Remove extra blank lines.

File: account_invoice_date/views/account_move_view.xml
Comment (dalonsod) [inline account_invoice_date/views/account_move_view.xml]: Missing empty line at file end

File: account_invoice_date/wizard/account_voucher_proforma_date.py
Comment (dalonsod) [inline account_invoice_date/wizard/account_voucher_proforma_date.py:8]: Unused import and variable, and `api` not used

File: account_invoice_date/wizard/account_voucher_proforma_date.py
Comment (dalonsod) [inline account_invoice_date/wizard/account_voucher_proforma_date.py]: `string` attribute in english

File: account_invoice_date/__manifest__.py
Comment (dalonsod) [inline account_invoice_date/__manifest__.py]: Use `summary` instead of description

**Review by dalonsod — CHANGES_REQUESTED** [review]
Thanks for the contribution!
Tested in runboat, AFAIK works. Code review, in general OCA conventions for new addons should be followed and `pre-commit run -a` command seems to be missing, see comments.

File: account_invoice_date/__manifest__.py
Comment (dalonsod) [inline account_invoice_date/__manifest__.py]: License not coherent with file header

Comment (jans23) [reply account_invoice_date/__manifest__.py]: BTW, we don't have a strong preference. Does OCA prefer any particular license or use by default?

File: account_invoice_date/README.rst
Comment (ikapasi-initos) [reply account_invoice_date/README.rst:1]: Hi, @dalonsod when I remove the " README.rst" file and then I've run the "pre-commit run -a" command faced an issue in the " README.rst" file. Please find the attached image.
README_issue

**Comment by ikapasi-initos** [comment]
Hi,@dalonsod I've implemented the feedback you can review it.

**Comment by ikapasi-initos** [comment]
Hi, @BT-anieto it would be great if you add reviews here

File: account_invoice_date/README.rst
Comment (AaronHForgeFlow) [inline account_invoice_date/README.rst]:
```suggestion
is set to current date. If an invoice is validated from Draft to Open manually, a window is displayed to query the invoice date from the user.
```

Can you put this in the first comment of the PR? this is a new module. People should know what is this about without looking at the code. Thanks.

File: account_invoice_date/__manifest__.py
Comment (AaronHForgeFlow) [reply account_invoice_date/__manifest__.py]: @jans23 Both are accepted and it is the author choice. Tipically it is AGPL unless it is a base module that is meant to be used in many other places. I think this module will not be used as dependency for others so APGL is fine.

File: account_invoice_date/wizard/account_voucher_proforma_date_view.xml
Comment (AaronHForgeFlow) [inline account_invoice_date/wizard/account_voucher_proforma_date_view.xml]:
```suggestion
                        string="Please enter invoice date"
```
I would avoid the part between brackets, but it is not blocking.

**Review by AaronHForgeFlow — COMMENTED** [review]
Functional review works fine

File: account_invoice_date/wizard/account_voucher_proforma_date_view.xml
Comment (ikapasi-initos) [reply account_invoice_date/wizard/account_voucher_proforma_date_view.xml]: Hi, @AaronHForgeFlow I've implemented the changes.

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
if the invoice date is already set, I would not show the wizard, just post the invoice as usual:

**Comment by jans23** [comment]
> if the invoice date is already set, I would not show the wizard, just post the invoice as usual: 

Good idea. That should also work for our workflow.

**Comment by ikapasi-initos** [comment]
@AaronHForgeFlow Could you please review it?

**Comment by dsolanki-initos** [comment]
Hi, @max3903 Could you please review this PR?

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
The invoice date is set, but it is still telling me to add a date:

image

Also, I would put in the button the same text I see in the invoice, "Confirm" or "Confirm and see invoice", I think that is less confusing for users. It works good overall.

**Comment by dsolanki-initos** [comment]
> The invoice date is set, but it is still telling me to add a date:
> 
> image
> 
> Also, I would put in the button the same text I see in the invoice, "Confirm" or "Confirm and see invoice", I think that is less confusing for users. It works good overall.

Sure, I will update the changes.

**Comment by dsolanki-initos** [comment]
@AaronHForgeFlow, I've updated the changes, Please review it.

**Comment by AaronHForgeFlow** [comment]
When the date is set manually it seems to be overwritten, that is not correct:
account_invoice_date_2

**Comment by AaronHForgeFlow** [comment]
Could you add at least one unit test that checks the main flow is correct? Thank you!

**Comment by dsolanki-initos** [comment]
> Could you add at least one unit test that checks the main flow is correct? Thank you!

Sure, I'll add the test case

**Comment by dsolanki-initos** [comment]
> Could you add at least one unit test that checks the main flow is correct? Thank you!

@AaronHForgeFlow, I've added the test cases kindly review them. Thank you

**Review by AaronHForgeFlow — APPROVED** [review]
Functional review

File: account_invoice_date/views/account_move_view.xml
Comment (dsolanki-initos) [reply account_invoice_date/views/account_move_view.xml]: An added empty line.

File: account_invoice_date/security/ir.model.access.csv
Comment (dsolanki-initos) [reply account_invoice_date/security/ir.model.access.csv]: removed extra lines

File: account_invoice_date/wizard/account_voucher_proforma_date.py
Comment (dsolanki-initos) [reply account_invoice_date/wizard/account_voucher_proforma_date.py:8]: Removed unnecessary imports

File: account_invoice_date/wizard/account_voucher_proforma_date.py
Comment (dsolanki-initos) [reply account_invoice_date/wizard/account_voucher_proforma_date.py]: Added the English string.

File: account_invoice_date/__manifest__.py
Comment (dsolanki-initos) [reply account_invoice_date/__manifest__.py]: Replaced description with a summary.

**Comment by dsolanki-initos** [comment]
Hello @dalonsod,  In this PR all the changes as per your feedback have been done, Kindly review it.

**Comment by dalonsod** [comment]
I still see `de.po` file. Remove it and I'll approve it. Thanks for your patience!

**Comment by jans23** [comment]
I don't understand. Why should the German translation be removed? Is there an alternative to the `de.po` file to provide translation?

**Comment by dalonsod** [comment]
Well, for new modules adding `po` files is actually allowed (https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#translations). Alternative, as usual, is Weblate once addon is merged.

**Review by dalonsod — APPROVED** [review]

File: account_invoice_date/readme/USAGE.rst
Comment (AaronHForgeFlow) [inline account_invoice_date/readme/USAGE.rst]:
```suggestion
#. Select the date and Click on "Confirm" button.
```

Minor thing before the merge

Comment (jans23) [reply account_invoice_date/readme/USAGE.rst]: Also it should be "click".

Comment (dsolanki-initos) [reply account_invoice_date/readme/USAGE.rst]: @AaronHForgeFlow, Changes are done, Kindly review it.

**Comment by AaronHForgeFlow** [comment]
LGTM

**Comment by jans23** [comment]
@AaronHForgeFlow This PR is approved and read-to-merge. What can we do to get it merged?

**Review by rafaelbn — COMMENTED** [review]
> When an customer invoice or vendor bill is validated from Draft to Open automatically, the invoice/bill date is set to current date. If an invoice/bill is validated from Draft to Open manually, a window is displayed to query the invoice/bill date from the user.

The name of the module IMHO is incorrecto as indicate that you are adding fields date in invoices

@AaronHForgeFlow could you say a better name like `account_invoice_date_update` or something similar?

Thank you all! 😄

**Comment by jans23** [comment]
We will happily change the module's name to whatever name you agree on.

**Pull request closed without merging**
